chore/setup database

osammy opened the pull request:
### Description ?
- Create a database connection module.
#### Type of change
- Chore
#### How Has This Been Tested?
- Invoked the database connection function and got a response
#### Checklist
- setup database for testing
- setup database for development
- setup database for staging
- setup database for production

osammy committed:
chore(backend): update Readme with database connection flow

osammy committed:
chore(backend): import database connection module from app.js

osammy committed:
feat(backend): export a database connection module

abidex4yemi commented on database/index.js:
```diff
@@ -0,0 +1,36 @@
+
+    mongoose.connect(
+      url,
+      { useNewUrlParser: true, useUnifiedTopology: true },
```
Good job on handling deprecation warning 👍

abidex4yemi commented on database/index.js:
```diff
@@ -0,0 +1,36 @@
+        break;
+
+      default:
+        url = process.env.DATABASE_URL_TEST;
```
The default connection string should be the development

twumm replied:
Can we have a local mongodb as well?
So
url = process.env.DATABASE_URL_TEST || 'mongodb://localhost/comcaldb' for running mongo locally

osammy replied:
@twumm 
 Yes, but you will need to set up a db locally on your machine. view the readme. I will be updating it

abidex4yemi commented on database/index.js:
```diff
@@ -0,0 +1,36 @@
+        url = process.env.DATABASE_URL_PRODUCTION;
+        break;
+
+      default:
```
This should be moved to config folder `config/databaseConfig.js`

osammy replied:
what should be moved?

osammy replied:
Okay let me do that

abidex4yemi replied:
The database configuration code

abidex4yemi replied:
Line 4 could you please indicate why you are returning promises

osammy replied:
@abidex4yemi I am using a promise because, sometimes especially in cases where you need to spin up a database instance like a seperate connection e.g in cron jobs, some operations require that the database connection be made before those operations are carried out. say for instance you want to delete an old events(I know we are not doing this, its just an example), you will need to have established a database connection before deleting, other wise there will be some error

abidex4yemi commented on app.js:
```diff
@@ -1,21 +1,29 @@
+  } catch (e) {
+     console.log('Failed to connect to database');
+  }
+})();
```
This should be moved to the database folder `database/index.js` and wrap in a function may be called `connectDB` that we could be called in our entry point file.

osammy replied:
I am not sure why this is necessary, because i could easily just call db() in app.js and that will suffice

osammy replied:
Are you sure this is a required step ?

abidex4yemi replied:
One of the reasons why I ask for the changes is because you use `IIFE` which I think is not necessary and the `conn` variable is not used anywhere in this file and the block doesn't explicitly say what it does

abidex4yemi replied:
Thinking of big-picture what if the first connection to the database failed due to network issue and you are asked to write a code that can make the request again how will that affect the length of the code block

abidex4yemi replied:
Remember `simple` is better than `complex`

osammy replied:
I get you, i only used an IIFE because i wanted to use an async and had to wrap in an async function. Anyways, but you can always call db() but nonetheless i will add the connection. Thanks for the feedback

abidex4yemi reviewed (changes requested):
Please make changes as per request, kindly indicate any suggestion

twumm commented on app.js:
```diff
@@ -1,21 +1,29 @@
 const express = require('express');
+require("dotenv").config();
```
dotenv requirement looks good.
Maybe we can have it on the first line to explicitly indicate that it runs before any other requirement.

osammy replied:
Eagle eyes!
thanks @twumm

abidex4yemi reviewed (changes requested):
You have done an amazing on setting up multiple database environments 💪I think we have to rearrange some code, I left some comment on the PR

osammy committed:
refactor(backend): refactor the database setup
Add a connect module in the /database folder
Created a dbConfig file to store all database configurations
Replaced the IIFE in app.js with a call to the dbConnection module
Updatad the README.md to include the default local database connection url

osammy commented:
I have Made adjustments as per the reviews stated by @twumm  and @abidex4yemi . pls review the changes and get back to me. Thanks.

osammy committed:
refactor(backend): deleted the connect.js file since it was redundant

osammy committed:
refactor(backend): refactored the app.js to connect without connect.js

osammy committed:
refactor(backend): modify dbconfig to return db connection url

osammy committed:
refactor(backend): consume the connection url from dbconfig

abidex4yemi reviewed (approved):
Everything looks good to me, good job on the refactor. 💪

osammy committed:
refactor(backend): removed res.render from app.js

osammy committed:
refactor(backend): remove a rouge console.log

osammy committed:
fix(backend): fix a bug with the morgan middleware
